src/uct/ib/rc/accel/rc_mlx5.inl

```diff
@@ -160,30 +160,92 @@ uct_rc_mlx5_iface_release_srq_seg(uct_rc_mlx5_iface_common_t *iface,
     ++iface->super.rx.srq.available;
 }
 
+#define uct_rc_mlx5_iface_mp_hash_lookup(_h_name, _h_ptr, _key, _last, _flags, \
+                                         _iface) \
+    ({ \
+        uct_rc_mlx5_mp_context_t *ctx; \
+        khiter_t h_it; \
+        int ret; \
+        h_it = kh_get(_h_name, _h_ptr, _key); \
+        if (h_it == kh_end(_h_ptr)) { \
+            /* No data from this sender - this must be the first fragment */ \
+            *(_flags) |= UCT_CB_PARAM_FLAG_FIRST; \
+            if (ucs_likely(_last)) { \
+                /* fast path - single fragment message */ \
+                return &(_iface)->tm.mp.dummy_ctx; \
+            } \
+            h_it = kh_put(_h_name, _h_ptr, _key, &ret); \
```

Comment:
instead of doing get+put, do only put. and check ret to see if it was added or not

Reply:
No need to add element if it is single fragment. Also put operation seems to be more heavy